codec-redis/src/main/java/io/netty/handler/codec/redis/FixedRedisMessagePool.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final String[] DEFAULT_SIMPLE_STRINGS = {
+            "OK",
+            "PONG",
+            "QUEUED",
+    };
+
+    private static final String[] DEFAULT_ERRORS = {
+            "ERR",
+            "ERR index out of range",
+            "ERR no such key",
+            "ERR source and destination objects are the same",
+            "ERR syntax error",
+            "BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.",
+            "BUSYKEY Target key name already exists.",
+            "EXECABORT Transaction discarded because of previous errors.",
+            "LOADING Redis is loading the dataset in memory",
+            "MASTERDOWN Link with MASTER is down and slave-serve-stale-data is set to 'no'.",
+            "MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. " +
+            "Commands that may modify the data set are disabled. Please check Redis logs for details " +
+            "about the error.",
+            "NOAUTH Authentication required.",
+            "NOREPLICAS Not enough good slaves to write.",
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    /**
+     * A shared object for {@link FixedRedisMessagePool}.
+     */
+    public static final FixedRedisMessagePool INSTANCE = new FixedRedisMessagePool();
+
+    // internal caches.
+    private Map<ByteBuf, SimpleStringRedisMessage> byteBufToSimpleStrings;
+    private Map<String, SimpleStringRedisMessage> stringToSimpleStrings;
+    private Map<ByteBuf, ErrorRedisMessage> byteBufToErrors;
+    private Map<String, ErrorRedisMessage> stringToErrors;
+    private Map<ByteBuf, IntegerRedisMessage> byteBufToIntegers;
+    private LongObjectMap<IntegerRedisMessage> longToIntegers;
+    private LongObjectMap<byte[]> longToByteBufs;
+
+    /**
+     * Creates a {@link FixedRedisMessagePool} instance.
+     */
+    private FixedRedisMessagePool() {
+        byteBufToSimpleStrings = new HashMap<ByteBuf, SimpleStringRedisMessage>(DEFAULT_SIMPLE_STRINGS.length);
+        stringToSimpleStrings = new HashMap<String, SimpleStringRedisMessage>(DEFAULT_SIMPLE_STRINGS.length);
+        for (String message : DEFAULT_SIMPLE_STRINGS) {
+            ByteBuf key = Unpooled.unmodifiableBuffer(
+                    Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8))));
+            SimpleStringRedisMessage cached = new SimpleStringRedisMessage(message);
+            byteBufToSimpleStrings.put(key, cached);
+            stringToSimpleStrings.put(message, cached);
+        }
+
+        byteBufToErrors = new HashMap<ByteBuf, ErrorRedisMessage>(DEFAULT_ERRORS.length);
```

Comment:
consider `loadFactor` to avoid resize ... maybe size maps a bit larger to allow for larger hash code space?